pyramid/session.py

```diff
@@ -101,6 +108,77 @@ def signed_deserialize(serialized, secret, hmac=hmac):
 
     return pickle.loads(pickled)
 
+
+def check_csrf_origin(request):
```

Comment:
I would like to see this function more reusable in a similar vein to `check_csrf_token`. For example a default signature like `check_csrf_origin(request, trusted_origins=None, raises=True)` where trusted_origins overrides the usage of the setting.